src/NuGet.Clients/NuGet.PackageManagement.UI/Options/PackageSourceCheckedListBox.cs

```diff
@@ -88,6 +93,21 @@ protected override void OnDrawItem(DrawItemEventArgs e)
                         // [checkbox] Name
                         //            Source (italics)
 
+                        ImageAttributes attributes = new ImageAttributes
+                        {
+                            StructSize = Marshal.SizeOf(typeof(ImageAttributes)),
+                            ImageType = (uint)_UIImageType.IT_Icon,
+                            Format = (uint)_UIDataFormat.DF_WinForms,
+                            LogicalWidth = 16,
+                            LogicalHeight = 16,
+                            Flags = (uint)_ImageAttributesFlags.IAF_RequiredFlags
+                        };
+
+                        IVsImageService2 vsImageService = (IVsImageService2)Microsoft.VisualStudio.Shell.Package.GetGlobalService(typeof(SVsImageService));
+                        IVsUIObject uIObj = vsImageService.GetImage(KnownMonikers.StatusWarning, attributes);
```

Comment:
I was wondering if we should get the warning icon image only if the package source is HTTP. It is possible that I don't understand the whole context of how it works currently because I don't know much about WinForms or WPF.

Reply:
I'd go further and say we should store the image in a static variable. This method, `OnDrawItem`, is called once per package source. There's no need to 1. get the image service instance since Visual Studio services are singletons, and 2. get the known image, which I'm reasonable confident is also a singleton. Admittedly customers don't open the dialog box very often, but this is an easy perf improvement, and this is happening on the UI thread after all.